.github/workflows/release.yml

```
@@ -0,0 +1,84 @@
+name: Build and publish distribution
+
+# This workflow will be triggered whenever the CI workflow is triggered in main branch is completed or a release is published
+on:
+  workflow_run:
+    workflows: [ "CI" ]
+    branches: [ main ]
+    types:
+      - completed
+  release:
+    types: [ published ]
+
+jobs:
+  publish-test:
+    runs-on: "ubuntu-latest"
+    steps:
+      - name: Checkout source
+        uses: actions/checkout@v3
+      - name: Set up Python 3.11
+        uses: actions/setup-python@v4
+        with:
+          python-version: 3.11
+      - name: Install build dependencies
+        run: pip install build
+      - name: Build distributions
+        run: python -m build
+      - name: Publish package to PyPI
+        # This condition ensures the CI workflow in main branch was successful and this is indeed a release published action
+        if: github.repository == 'cleanlab/cleanvision' && github.event.workflow_run.conclusion == 'success' && github.event_name == 'release' && github.event.action == 'published'
```

Correct me if I'm wrong, @sanjanag, but aren't `workflow_run` and `release` disjoint events?

Is this the correct expression? (Double-check my placement of parentheses)

```suggestion
        if: github.repository == 'cleanlab/cleanvision' && ( github.event.workflow_run.conclusion == 'success' || ( github.event_name == 'release' && github.event.action == 'published' ) )
```

---

Yes, they are disjoint events. My reason for adding workflow_run to the condition was to ensure the `CI` workflow was successful in the `main` branch. With the OR condition, this will release on main in merge also, which I did not want, hence the AND condition.

---

I guess what is confusing me is: If the workflow_run condition is true, then the CI wasn't triggered by a release event, so the latter part of the expression will be false?

---

you are right, I can't place a check on disjoint events. Removed workflow_run since it was not serving the intended purpose.